tripal/tripal
# Unpublish UI

# Tripal 4 Core Dev Task

### Issue #1350 


### Tripal Version:  4.x

## Description


In order to implement the `ChadoStorage::deleteValues()` function we need to differentiate between an "unpublish" and  "delete".   This PR simply adds that functionality to the UI. It doesn't' fully implement a delete. or the `deleteValues()` function.  I separted this as a separte PR to simplify the number of files that have to be checked in a full PR for this issue.

## Testing?



1.  Go to a Tripal entity page. You'll see both an "Unpublish" and "Delete" tab at the to of the page.  Click and test each one. For the delete you should see a message that the delete is currently not fully implemented and it works like an "unpublish".
2. Edit any Tripal entity page. You'll see a new "Unpublish" button (it's red on the default theme) at the bottom of the page.  The delete button is the default button in Drupal.  They should also both work. They use the same underlying functions, but use a "pop-up" to confirm unpublish/delete (default Drupal behavior).
3. Go the the Tripal content page listing. In the action bar on the right you'll now see an unpublish option. It uses the same pop-up as the entity edit page.
